similarity-data-impl/src/main/java/org/phenotips/data/similarity/internal/DefaultPatientGenotype.java

```diff
@@ -90,31 +97,68 @@ public DefaultPatientGenotype(Patient patient)
     }
 
     /**
-     * Return a set of the names of candidate genes listed for the given patient.
+     * Return a set of the names of solved (candidate if no solved) genes listed for the given patient.
+     *
+     * @param p the {@link Patient}
+     * @return a (potentially-empty) unmodifiable set of the names of genes
+     */
+    private static Set<String> getManualGeneNames(Patient patient)
+    {
+        Set<String> geneNames = new HashSet<String>();
+       
+        PatientData<Map<String, String>> allGenes = patient.getData("genes");
+        if (allGenes != null && allGenes.isIndexed()) {
+            Set<String> geneCandidateNames = new HashSet<String>();
```

Comment:
Create two HashSets, one for solved and one for candidate gene names. Return the appropriate set at the end.